WordPress/src/test/java/org/wordpress/android/viewmodel/posts/PostListItemUiStateHelperTest.kt

```diff
@@ -72,10 +82,117 @@ class PostListItemUiStateHelperTest {
+        val state = createPostListItemUiState(post = createPostModel(isLocallyChanged = true))
+        assertThat(state.data.statusLabels).contains(UiStringRes(R.string.local_changes))
+    }
+
+    @Test
+    fun `version conflict label shown for posts with version conflict`() {
+        val state = createPostListItemUiState(unhandledConflicts = true)
+        assertThat(state.data.statusLabels).contains(UiStringRes(R.string.local_post_is_conflicted))
+    }
+
+    @Test
+    fun `uploading post label shown when the post is being uploaded`() {
+        val state = createPostListItemUiState(uploadStatus = createUploadStatus(isUploading = true))
+        assertThat(state.data.statusLabels).contains(UiStringRes(R.string.post_uploading))
+    }
+
+    @Test
+    fun `uploading media label shown when the post's media is being uploaded`() {
+        val state = createPostListItemUiState(uploadStatus = createUploadStatus(hasInProgressMediaUpload = true))
+        assertThat(state.data.statusLabels).contains(UiStringRes(R.string.uploading_media))
+    }
+
+    @Test
+    fun `queued post label shown when the post has pending media uploads`() {
+        val state = createPostListItemUiState(uploadStatus = createUploadStatus(hasPendingMediaUpload = true))
+        assertThat(state.data.statusLabels).contains(UiStringRes(R.string.post_queued))
+    }
+
+    @Test
+    fun `queued post label shown when the post is queued for upload`() {
+        val state = createPostListItemUiState(uploadStatus = createUploadStatus(isQueued = true))
+        assertThat(state.data.statusLabels).contains(UiStringRes(R.string.post_queued))
+    }
+
+    @Test
+    fun `error uploading media label shown when the media upload fails`() {
+        val state = createPostListItemUiState(
+                uploadStatus = createUploadStatus(uploadError = UploadError(MediaError(AUTHORIZATION_REQUIRED)))
+        )
+        assertThat(state.data.statusLabels).contains(UiStringRes(R.string.error_media_recover_post))
+    }
+
+    @Test
+    fun `error uploading post label shown when the post upload fails`() {
+        val errorMsg = "testing error message"
+        val state = createPostListItemUiState(
+                uploadStatus = createUploadStatus(uploadError = UploadError(PostError(GENERIC_ERROR, errorMsg)))
+        )
+        assertThat(state.data.statusLabels).contains(UiStringText(errorMsg))
+    }
+
+    @Test
+    fun `error label has precedence over info labels`() {
+        val state = createPostListItemUiState(
+                post = createPostModel(isLocallyChanged = true, status = POST_STATE_PRIVATE),
+                uploadStatus = createUploadStatus(uploadError = UploadError(MediaError(AUTHORIZATION_REQUIRED)))
+        )
+        assertThat(state.data.statusLabels).contains(UiStringRes(R.string.error_media_recover_post))
+        assertThat(state.data.statusLabels.size).isEqualTo(1)
+    }
+
+    @Test
+    fun `multiple info labels are being shown together`() {
```

Comment:
Should there be assertions for the order? Or the order doesn't matter?

Reply:
Good question. I believe the order doesn't matter. If we decide it matters, I'll make sure to add new tests.